app/views/certificates/pathways/_recommended_courses.html.erb

```diff
@@ -26,8 +26,12 @@
                 <%= activity_type(pa) %>
               </p>
               <p class="ncce-pathway-courses__location icon-clock govuk-!-margin-top-0">
-                <% course = Achiever::Course::Template.find_by_activity_code(pa.stem_activity_code)%>
-                <%= pluralize(course.duration_value, course.duration.downcase.chomp('s')) if course.duration.present? %> 
+                <% begin %>
```

Comment:
I’m never a huge fan of throwing and catching exceptions as it creates a bit of a nightmare of control flow. In my pathways branch, I’ve added a
```rb
  def self.maybe_find_by_activity_code(activity_code)
    templates = all
    templates.find { |val| val.activity_code == activity_code.upcase }
  end
```
method for potentially finding an activity.
Could you add this instead and handle the nil case?

Unfortunately sending that exception to sentry won’t tell us which CPDs are missing from dynamics because that find_by_activity_code method throws `raise ActiveRecord::RecordNotFound unless template` directly without any message.
What would be quite nice is if we could add a message to that not found, and maybe also add quiet sentry reporting to the maybe_find_by_activity_code method

Reply:
I've done all this, and re-based, except "add a message to that not found" (as I haven't thought of a useful message or a quick way to add it.) Pls re-review.